datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/KafkaPositionTracker.java

```diff
@@ -432,14 +435,30 @@ public static Builder builder() {
    * Note that the provided consumer must not be operated on by any other thread, or a concurrent modification condition
    * may arise.
    *
+   * Ideally the partitions provided to this method are owned all by the same broker for performance purposes.
+   *
    * Use externally for testing purposes only.
    */
   @VisibleForTesting
   void queryBrokerForLatestOffsets(@NotNull final Consumer<?, ?> consumer,
-      @NotNull final Set<TopicPartition> partitions) {
+      @NotNull final Set<TopicPartition> partitions, @NotNull final Duration requestTimeout) {
     for (final List<TopicPartition> batch : Iterables.partition(partitions, BROKER_OFFSETS_FETCH_SIZE)) {
+      LOG.debug("Fetching the latest offsets for {} partitions", batch.size());
+      LOG.trace("Fetching the latest offsets for partitions: {}", batch);
       final Instant queryTime = Instant.now();
-      final Map<TopicPartition, Long> offsets = consumer.endOffsets(batch);
+      final Map<TopicPartition, Long> offsets;
+      try {
+        offsets = consumer.endOffsets(batch, requestTimeout);
+      } catch (Exception e) {
+        LOG.warn("Unable to fetch latest offsets for {} partitions. Will retry later. Reason: {}", batch.size(), e.getMessage());
+        LOG.debug("Unable to fetch latest offsets for {} partitions. Will retry later.", batch.size(), e);
```

Comment:
why do you need to log in three different levels?

Reply:
I was trying to tune each log level to give differing degrees of data, but I agree that it's a little overkill. I think this really only needs to be at TRACE level, so I will make that change.